app/Services/UserService.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace App\Services;
+
+use App\Dao\UserDao;
+use App\Models\User;
+use Illuminate\Http\Request;
+use App\Contracts\Dao\UserDaoInterface;
+use App\Contracts\Services\UserServiceInterface;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Hash;
+
+class UserService implements UserServiceInterface
+{
+    protected $userDao;
+
+    public function __construct(UserDaoInterface $userDaoInterface)
+    {
+        $this->userDao = $userDaoInterface;
+    }
+
+    /**
+     * insert user
+     *
+     * @param Request $request
+     * @return null
+     */
+    public function insert(Request $request): null
```

Comment:
May be the return type is `void` if you don't return anything.

Reply:
when return type is changed to null, it throws the following error
"A void function must not return a value"

at 44
return $this->userDao->insert($insertData);